lib/spack/docs/workflows.rst

```diff
@@ -0,0 +1,1150 @@
+
+Filesystem views offer an alternative to environment modules, another
+way to assemble packages in a useful way and load them into a user's
+environment.
+
+A filesystem view is a single directory tree that is the union of the
+directory hierarchies of a number of installed packages; it is similar
+to the directory hiearchy that might exist under ``/usr/local``.  The
+files of the view's installed packages are brought into the view by
+symbolic or hard links, referencing the original Spack installation.
+
+When software is built and installed, absolute paths are frequently
+"baked into" the software, making it non-relocatable.  This happens
+not just in RPATHs, but also in shebangs, configuration files, and
+assorted other locations.
+
+Therefore, programs run out of a Spack view will typically still look
+in the original Spack-installed location for shared libraries and
+other resources.  This behavior is not easily changed; in general,
+there is no way to know where absolute paths might be written into an
+installed package, and how to relocate it.  Therefore, the original
+Spack tree must be kept in place for a filesystem view to work, even
+if the view is built with hardlinks.
+
+.. FIXME: reference the relocation work of Hegner and Gartung (PR #1013)
+
+
+""""""""""""""""""""""
+Using Filesystem Views
+""""""""""""""""""""""
+
+A filesystem view is created, and packages are linked in, by the ``spack
+view`` command's ``symlink`` and ``hardlink`` sub-commands.  The
+``spack view remove`` command can be used to unlink some or all of the
+filesystem view.
+
+The following example creates a filesystem view based
+on an installed ``cmake`` package and then removes from the view the
+files in the ``cmake`` package while retaining its dependencies.
+
+.. code-block:: console
+
+   $ spack view --verbose symlink myview cmake@3.5.2
+   ==> Linking package: "ncurses"
+   ==> Linking package: "zlib"
+   ==> Linking package: "openssl"
+   ==> Linking package: "cmake"
+
+   $ ls myview/
+   bin  doc  etc  include  lib  share
+
+   $ ls myview/bin/
+   captoinfo  clear  cpack     ctest    infotocap        openssl  tabs  toe   tset
+   ccmake     cmake  c_rehash  infocmp  ncurses6-config  reset    tic   tput
+
+   $ spack view --verbose --dependencies false rm myview cmake@3.5.2
+   ==> Removing package: "cmake"
+
+   $ ls myview/bin/
+   captoinfo  c_rehash  infotocap        openssl  tabs  toe   tset
+   clear      infocmp   ncurses6-config  reset    tic   tput
+
+.. note::
+
+    If the set of packages being included in a view is inconsistent,
+    then it is possible that two packages will provide the same file.  Any
+    conflicts of this type are handled on a first-come-first-served basis,
+    and a warning is printed.
+
+.. note::
+
+    When packages are removed from a view, empty directories are
+    purged.
+
+""""""""""""""""""
+Fine-Grain Control
+""""""""""""""""""
+
+The ``-e`` and ``-d`` option flags allow for fine-grained control over
```

Comment:
You should use the long option names wherever possible (`--exclude` and `--dependencies` in this case).